tests/test_patchanalysis.py

```diff
@@ -36,12 +38,22 @@ def assertWarnings(self, warnings, expected_warnings):
         self.assertEqual(len(missed_warnings), 0)
         self.assertEqual(len(unexpected_warnings), 0)
 
-    def assertDictEqual(self, d1, d2, msg=None):
```

Comment:
Can you leave this override in place? It should give a more undestandable error when something fails.